vignettes/howto-dm-rows.Rmd

```diff
@@ -47,28 +47,28 @@ With the changes confirmed, you execute the method again, this time with the arg
 Note that `in_place = FALSE` is the default: you must opt in to actually change data on the database.
 
 Each method has its own requirements in order to maintain database consistency.
-These involve constraints on primary key values as they are how rows are identified.
+These involve constraints on primary key values that uniquely identify rows.
 
 | Method | Requirements |
 |--------|--------------|
-| `dm_rows_insert()` | The primary keys must differ from existing records.|
+| `dm_rows_insert()` | Primary keys must be present for all tables.|
+| `dm_rows_append()` | Primary keys must differ from existing records.|
 | `dm_rows_update()` | Primary keys must match for all records to be updated.|
 | `dm_rows_patch()` | Updates missing values in existing records. Primary keys must match for all records to be patched.|
 | `dm_rows_upsert()` | Updates existing records and adds new records, based on the primary key.|
-| `dm_rows_delete()` | Removes matching records based on the primary key.|
-| `dm_rows_truncate()` | Removes all records, only for tables in the changeset dm.|
+| `dm_rows_delete()` | Removes matching records based on the primary key. Primary keys must match for all records to be deleted.|
 
 To ensure the integrity of all relations during the process, all methods automatically determine the correct processing order for the tables involved.
-For operations that create records, parent tables are processed before child tables.
-For `dm_rows_delete()` and `dm_rows_truncate()`, child tables are processed before their parent tables.
+For operations that create records, parent tables (which hold primary keys) are processed before child tables (which hold foreign keys).
+For `dm_rows_delete()`, child tables are processed before their parent tables.
 For more details on this see `vignette("howto-dm-theory")` and `vignette("howto-dm-db")`.
 
```

Comment:
Should we mention that the user is still responsible for setting transactions to ensure integrity of operations across multiple tables?

Reply:
Good idea. Added this note.